astropy/coordinates/tests/test_intermediate_transformations.py

```diff
@@ -631,3 +634,79 @@ def test_tete_transforms():
     with pytest.warns(AstropyDeprecationWarning, match='The use of'):
         tete_alt = _apparent_position_in_true_coordinates(moon)
     assert_allclose(tete_coo1.separation_3d(tete_alt), 0*u.mm, atol=100*u.mm)
+
+
+def test_straight_overhead():
+    """
+    With a precise CIRS<->AltAz transformation this should give Alt=90 exactly
+
+    If the CIRS self-transform breaks it won't, due to improper treatment of aberration
+    """
+    t = Time('J2010')
+    obj = EarthLocation(-1*u.deg, 52*u.deg, height=10.*u.km)
+    home = EarthLocation(-1*u.deg, 52*u.deg, height=0.*u.km)
+
+    # An object that appears straight overhead - FOR A GEOCENTRIC OBSERVER.
+    # Note, this won't be overhead for a topocentric observer because of
+    # aberration.
+    cirs_geo = obj.get_itrs(t).transform_to(CIRS(obstime=t))
+
+    # now get the Geocentric CIRS position of observatory
+    obsrepr = home.get_itrs(t).transform_to(CIRS(obstime=t)).cartesian
+
+    # topocentric CIRS position of a straight overhead object
+    cirs_repr = cirs_geo.cartesian - obsrepr
+
+    # create a CIRS object that appears straight overhead for a TOPOCENTRIC OBSERVER
+    topocentric_cirs_frame = CIRS(obstime=t, location=home)
+    cirs_topo = topocentric_cirs_frame.realize_frame(cirs_repr)
+
+    aa = cirs_topo.transform_to(AltAz(obstime=t, location=home))
+    assert_allclose(aa.alt, 90*u.deg)
+
+
+@pytest.mark.remote_data
+def test_aa_high_precision():
+    """
+    These tests are provided by @mkbrewer - see issue #10356.
+
+    The code that produces them agrees very well (<0.5 mas) with SkyField once Polar motion
```

Comment:
Wasn't SkyField just updated?

Reply:
The SkyField update is still a work in progress at this time. The basic ITRS<->TIRS transforms are tested and working, but there is a bit of infrastructure that still needs doing, so it'll be a few weeks.